agent/directory.go

```diff
@@ -22,7 +23,7 @@ func (a *Agent) createTaskDirectory(tc *taskContext, taskConfig *model.TaskConfi
 	h := md5.New()
 
 	_, err := h.Write([]byte(
-		fmt.Sprintf("%s_%d_%d", taskConfig.Task.Id, taskConfig.Task.Execution, os.Getpid())))
+		fmt.Sprintf("%s_%d_%d_%f", taskConfig.Task.Id, taskConfig.Task.Execution, os.Getpid(), rand.Float64())))
```

Comment:
this makes task directory not reproducible, which doesn't seem like a huge liability, but it does sort of make the whole hashing thing less useful? why not just use a UUID at that point? 

Why don't we make the test fixtures clean things up better?